java/vector/src/main/java/org/apache/arrow/vector/IntVector.java

```diff
@@ -344,8 +345,9 @@ public TransferPair makeTransferPair(ValueVector to) {
   }
 
   @Override
-  public void setEncodedValue(int index, int value) {
-    this.setSafe(index, value);
+  public void setWithPossibleTruncate(int index, long value) {
+    Preconditions.checkArgument(value <= Integer.MAX_VALUE, "value is overflow: %s", value);
```

Comment:
@jacques-n where you thinking we would silently truncate here instead of throwing an exception?

Reply:
yes, silent truncate

Reply:
remove all Preconditions.checkArgument? @emkornfield

Reply:
yes, that is what the name of the method implies.   We can see how often this leads to bugs and potentially add some warnings.

Reply:
Fine, fixed now.